### DISCUSSION TITLE
Explicitly skipping completion scripts.

### DISCUSSION THREAD

**Discussion by jasonkarns** [Homebrew/discussions#3927] [Everyday usage] [answered]
Initial issue:

I'm setting up a new computer (scripted) using a Brewfile. This includes a formula that I explicitly _want_ to install (`hub`). However, after installation, shell completion for `git` was broken. It took nearly an hour of debugging to uncover that hub's shell completion script is currently incompatible/broken/buggy with recent versions of git's completion. What's interesting to note about this, is that at some point in the past on my old machine, I _knew_ this and had deleted hub's completion symlink from the bash completion directory. (ie, `BASH_COMPLETION_COMPAT_DIR` for bash-completion@2 which homebrew symlinks completion scripts to via: `$(brew --prefix)/etc/bash_completion.d`) However, because hub was listed in the Brewfile, it was installed on the new machine and (to my knowledge) there is no mechanism to instruct brew-bundle to skip a particular formula's completion script.

So there are two related questions I have.

Firstly, is there any mechanism (like a `--no-completion` flag) that can be passed to `brew install hub` that would instruct brew to skip symlinking of the completion scripts?

Secondly, and likely dependent on the first, would there be any mechanism for brew-bundle to lean on that feature so a Brewfile can be constructed that replicates that installation option?

**Comment by SMillerDev** [comment, marked as answer]
No, we don't support mechanisms for partial installation. The best solution is to file an issue upstream and help the `hub` developers resolve the problem.
